src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs

```diff
@@ -309,6 +311,21 @@ private void EnsureCollationIsSupportedIfAnyRequestHasCollation(RetryableWriteCo
             }
         }
 
+        private void EnsureHintIsSupportedIfAnyRequestHasHint(RetryableWriteContext context)
+        {
+            var serverVersion = context.Channel.ConnectionDescription.ServerVersion;
+            if (!Feature.HintForUpdateAndReplaceOperations.IsSupported(serverVersion, out var isExceptionAllowed) && isExceptionAllowed)
```

Comment:
Using the inherited `IsSupportedMethod` and the suggested `DriverMustThrowIfNotSupported` method lines 316-317 can become:

    var hintFeature = Feature.HintForUpdateAndReplaceOperations;
    var serverVersion = context.Channel.ConnectionDescription.ServerVersion;
    if (!hintFeature.IsSupported(serverVersion) && hintFeature.DriverMustThrowIfNotSupported(serverVersion))

Similarly for the other places that perform the same test.

Reply:
I used only `DriverMustThrowIfNotSupported`.
Because `!hintFeature.IsSupported(serverVersion)` returns true for servers < 4.2 and `hintFeature.DriverMustThrowIfNotSupported(serverVersion)` returns true for servers < 3.6. Since they are used in conjunction, I left only `DriverMustThrowIfNotSupported` check. That would be sufficient, unless we plan to change this logic soon.
@rstam Could you please conirm you are ok with this change?

Reply:
Yes it's OK. I like it.